readthedocs/oauth/migrations/0013_data_migration_for_remote_relation_model.py

```diff
@@ -0,0 +1,49 @@
+# Generated by Django 2.2.16 on 2020-10-05 06:15
+
+import json
+from itertools import islice
+
+from django.db import migrations
+
+
+def move_data_to_remote_relations(apps, schema_editor):
+    RemoteRelation = apps.get_model('oauth', 'RemoteRelation')
+
+    def remote_relations_generator(relations):
+        for relation in relations:
+            relation.account = relation.remoterepository.account
+            relation.active = relation.remoterepository.active
+            relation.admin = relation.remoterepository.admin
+            relation.pub_date = relation.remoterepository.pub_date
+            try:
+                relation.json = json.loads(relation.remoterepository.json)
+            except json.decoder.JSONDecodeError:
+                pass
+
+            yield relation
+
+    relations_queryset = RemoteRelation.objects.all().select_related('remoterepository')
+    remote_relations = remote_relations_generator(relations_queryset)
+    batch_size = 5000
```

Comment:
Not sure about what the size can be for each batch. but if we do all of them at once it will create a huge SQL Query

Reply:
Yeah, this is one of my concerns of this migrations. I don't have a good answer yet and we may ask somebody else for advice here.

The `RemoteRepostory` table is incredible huge. I already had to trigger the creation of an INDEX in this table manually without blocking things (concurrently) because it took hours.

In this case, we will need to find a way to not block the DB while deploying these changes.